### PR TITLE
More tests for ContextExtension semantics

### PR DESCRIPTION
**Pull request by kushti**
In this PR there are a couple of tests for validation of transactions with negative-id context extension variables

### CODE REVIEW

File: src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala [hunk 1]
```diff
@@ -543,4 +543,46 @@ class ErgoTransactionSpec extends ErgoPropertyTest with ErgoTestConstants {
     }
   }
 
+  property("context extension with neg id") {
+    val negId: Byte = -1
+
+    ADKey @@ Base16.decode("c95c2ccf55e03cac6659f71ca4df832d28e2375569cec178dcb17f3e2e5f7742").get
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala:549]: This probably should be an explicit assertion.

File: src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala [hunk 2]
```diff
@@ -543,4 +543,46 @@ class ErgoTransactionSpec extends ErgoPropertyTest with ErgoTestConstants {
...
+      .map(ErgoTransaction.apply)
+
+    txTry.isSuccess shouldBe false
+  }
+
+
+  property("context extension with neg and pos ids") {
+    val negId: Byte = -20
+
+    ADKey @@ Base16.decode("c95c2ccf55e03cac6659f71ca4df832d28e2375569cec178dcb17f3e2e5f7742").get
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala:570]: This probably should be an explicit assertion.

File: src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala [hunk 3]
```diff
@@ -543,4 +543,46 @@ class ErgoTransactionSpec extends ErgoPropertyTest with ErgoTestConstants {
...
+    val input = Input(b.id, ProverResult(Array.emptyByteArray, ContextExtension(Map(negId -> IntConstant(0)))))
+
+    val oc = new ErgoBoxCandidate(b.value, b.ergoTree, b.creationHeight)
+
+    val utx = UnsignedErgoTransaction(IndexedSeq(input), IndexedSeq(oc))
+
+    val txTry = defaultProver.sign(utx, IndexedSeq(b), IndexedSeq.empty, emptyStateContext)
+      .map(ErgoTransaction.apply)
+
+    txTry.isSuccess shouldBe false
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala:563]: I think it is better to test the exact reason of non-success.

File: src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala (hunk 1 above)
Comment (kushti) [reply src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala]: removed

File: src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala (hunk 3 above)
Comment (kushti) [reply src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala:561]: done

**Pull request merged**